### ISSUE TITLE
ForbidUnusedMatchResultRule false positive

### ISSUE THREAD

**Issue by xificurk** [shipmonk-rnd/phpstan-rules#170]
The rule reports false positive for construct:

```
$string .= match (/* ... */) {/* ... */};
```
